openmetadata-ui/src/main/resources/ui/src/components/DataProductSelectForm/DataProductsSelectForm.test.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { fireEvent, render } from '@testing-library/react';
+import React from 'react';
+import DataProductsSelectForm from './DataProductsSelectForm';
+
+const mockOnSubmit = jest.fn();
+const mockOnCancel = jest.fn();
+
+describe('Test Suite Form Page', () => {
+  it('renders without errors', () => {
+    const { getByTestId } = render(
+      <DataProductsSelectForm
+        defaultValue={[]}
+        fetchApi={() => Promise.resolve({ data: [], paging: { total: 0 } })}
+        placeholder="Select products"
+        onCancel={mockOnCancel}
+        onSubmit={mockOnSubmit}
+      />
+    );
+
+    // Ensure that the component renders without errors
+    expect(getByTestId('data-product-selector')).toBeInTheDocument();
+  });
+
+  it('calls onCancel function when Cancel button is clicked', () => {
+    const { getByTestId } = render(
+      <DataProductsSelectForm
+        defaultValue={[]}
+        fetchApi={() => Promise.resolve({ data: [], paging: { total: 0 } })}
+        placeholder="Select products"
+        onCancel={mockOnCancel}
+        onSubmit={mockOnSubmit}
+      />
+    );
+
+    const cancelButton = getByTestId('cancelAssociatedTag');
+    fireEvent.click(cancelButton);
```

Comment:
I recommend using userEvent for testing user interaction. for more info, you can read [here ](https://testing-library.com/docs/user-event/intro/#differences-from-fireevent)